src/edu/washington/escience/myria/HdfsTupleWriter.java

```diff
@@ -0,0 +1,64 @@
+package edu.washington.escience.myria;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+
+import edu.washington.escience.myria.storage.ReadableTable;
+
+/**
+ * HdfsTupleWriter is a {@link TupleWriter} that writes tuples to HDFS.
+ */
+public class HdfsTupleWriter implements TupleWriter {
+
+  /** Required for Java serialization. */
+  static final long serialVersionUID = 1L;
+
+  /** The {@link OutputStream} to which we write the data. **/
+  private FSDataOutputStream outStream;
+
+  /**
+   * Constructs a {@link HdfsTupleWriter}.
+   */
+  public HdfsTupleWriter() {
+  }
+
+  @Override
+  public void open(final OutputStream stream) {
+    outStream = (FSDataOutputStream) stream;
+  }
+
+  @Override
+  public void writeColumnHeaders(final List<String> columnNames) throws IOException {
+  }
+
+  @Override
+  public void writeTuples(final ReadableTable tuples) throws IOException {
+    /* Write each row to the output stream */
+    for (int i = 0; i < tuples.numTuples(); ++i) {
+      String row = "";
+      for (int j = 0; j < tuples.numColumns(); ++j) {
+        row += tuples.getObject(j, i).toString();
+        row += j == tuples.numColumns() - 1 ? '\n' : ',';
```

Comment:
How does this handle embedded commas/newlines? It might be more robust to use an existing CSV formatter to handle these issues, or to use control characters for the field/record delimiters (e.g., Hive uses Ctrl-A (`\001`) as its default field delimiter). `CSVFormat` in `CsvTupleWriter` is pretty heavily customizable (you'd need to expose the `CSVFormat` field in a public constructor).